src/rust/rust-proto-new/src/graplinc/grapl/pipeline/v1beta1.rs

```diff
@@ -42,23 +42,23 @@ impl TryFrom<MetadataProto> for Metadata {
     fn try_from(metadata_proto: MetadataProto) -> Result<Self, Self::Error> {
         let tenant_id = metadata_proto
             .tenant_id
-            .ok_or(SerDeError::MissingField("tenant_id".to_string()));
+            .ok_or_else(|| SerDeError::MissingField("tenant_id".to_string()));
```

Comment:
Yes, it was called out by a clippy lint. `ok_or_else` lazily evaluates the `.to_string()`.

Reply:
Yeah... it's not quite as glaring in this case, but it's really important for function calls that consume a lot of resources; you only want to invoke that overhead if there's actually an error to deal with. Clippy just says "hey, you've got a function invocation in here; you probably want to do that lazily".

(If the error variant could take a `&str`, then it could have stayed as `ok_or`, I believe.)

Reply:
Would it make more sense to have the variant take a `&'static str`?

Reply:
If that works and doesn't introduce a bunch of complexity, it may be worthwhile. The code as it exists isn't bad, though.